File: src/mordor_rp.py
from requests import get

class MordorRP:
	def __init__(self) -> None:
		self.api = "http://mordormail.ru/launcher"
		self.headers = {
			"user-agent": "okhttp/3.12.12"
		}

	def get_online(self) -> list:
		return get(
			f"{self.api}/monitoring/online.php",
			headers=self.headers).json()

	def get_protect_files(self) -> list:
		return get(
			f"{self.api}/protect/protect_file.json",
			headers=self.headers).json()

	def get_config(self) -> dict:
		return get(
			f"{self.api}/config/config_mordor.json",
			headers=self.headers).json()
		

File: src/mordor_rp.js
class MordorRp {
	constructor() {
		this.api = "http://mordormail.ru/launcher"
		this.headers = {
			"user-agent": "okhttp/3.12.12"
		}
	}

	async getOnline() {
		const response = await fetch(
			`${this.api}/monitoring/online.php`, {
				method: "GET",
				headers: this.headers
			})
		return response.json()
	}

	async getProtectFiles() {
		const response = await fetch(
			`${this.api}/protect/protect_file.json`, {
				method: "GET",
				headers: this.headers
			})
		return response.json()
	}

	async getConfig() {
		const response = await fetch(
			`${this.api}/config/config_mordor.json`, {
				method: "GET",
				headers: this.headers
			})
		return response.json()
	}
}

module.exports = {MordorRp}
